README.md

```diff
@@ -730,6 +730,8 @@ In order to achieve this kind of workflow, all you need to do is specify the dep
 codePush.sync({ deploymentKey: userProfile.CODEPUSH_KEY });
 ```
 
+*NOTE: Deployment key specified by `codePush.sync({ deploymentKey: key })` will not persist across app restarts (user kills app process and launch it again) and will be taken from native-side on next app launch. In case app restarts programmatically (while using IMMEDIATE installation mode) deployment key will persists.*
```

Comment:
>I'm not sure what is meant by persisting after the app restarts programatically.

`IMMEDIATE` installation mode uses `RestartManager.restartApp()` method to restart app programatically (user does not kill the app process and launch it again)

>I'm not sure that using IMMEDIATE install mode or not makes any difference - are you saying that the value set to the variable key is lost when the app restarts, but not when it restarts programmatically? Because that's not what I'd expect.

I've created demo app and could confirm, that the value set to variable key is not probably lost when app restart programmatically. Here is the code I've used to reproduce the issue:
```javascript
  syncProduction() {
    CodePush.sync(
      { deploymentKey: "iDKl3x098ODmeItIQEyEMjV3M7EjVyM2ZvIuz" } // Production deployment key
    )
  }

  syncStaging() {
    CodePush.sync(); // Staging deployment key should be taken from native-side
  }

  restartApp() {
    CodePush.restartApp(); // both "CodePush.restartApp()" and "IMMEDIATE" installation mode are using "RestartManager.restartApp()" method to restart app programatically (user does not kill the app process and launch it again)
  }
```
Reproduction steps:
1. call `syncProduction()` method
2. call `restartApp()` method
3. call `syncStaging()` method

**Actual**: CodePush prints "App up to date." due to probably using Production deployment key
**Expected**: CodePush prints "Update installed and will be applied on restart." due to using staging deployment key

I believe I should fix this confusing behavior instead of updating the docs. I suppose that the expected behavior is `*NOTE: The deployment key specified via `codePush.sync({ deploymentKey: key })` does not persist beyond the given call, and needs to be specified as an option each time it is used.*` as you've described above.